File: lessons/compiled_lang/java/src/f03.loop.java
// testbox: title="indexed for loop over List"
// Not public - see a00.output.java for why.
import java.io.File;
import java.util.ArrayList;
import java.util.Collections;
import java.util.List;

class F03Loop {
    public static void main(String[] args) {
        File dir = new File("dirtest");
        List<String> items = new ArrayList<>();
        for (String name : dir.list()) items.add(name);
        Collections.sort(items);

        // indexed for loop
        for (int i = 0; i < items.size(); i++) {
            String item = items.get(i);
            if (new File(dir, item).isDirectory())
                System.out.println(item + " is a directory");
            else
                System.out.println(item + " is not a directory");
        }
    }
}

File: lessons/compiled_lang/rust/src/f03.loop.rs
// testbox: title="indexed for-in over a Range"
use std::fs;

fn main() {
    let mut entries: Vec<_> = fs::read_dir("dirtest").unwrap().map(|e| e.unwrap()).collect();
    entries.sort_by_key(|e| e.file_name());

    // indexed access - for-in over a Range, indexing into the Vec
    for i in 0..entries.len() {
        let entry = &entries[i];
        let name = entry.file_name().to_string_lossy().to_string();
        if entry.path().is_dir() {
            println!("{} is a directory", name);
        } else {
            println!("{} is not a directory", name);
        }
    }
}
